### PR TITLE
feat: remove code backticks in CLI help messages

### LINKED ISSUE #303: Remove backticks from generated help messages
### What needs to get done

Parse out the extra backticks from the auto-generated help messages.

### Why it needs to get done

Currently, generated help messages preserve the double backticks (``) from reST documentation when printing help messages to the terminal. These look out of place and muddy up the appearance of the help messages.

### PR DESCRIPTION
**Pull request by bepri**
- [x] Have you followed the guidelines for contributing?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] Have you successfully run `tox`?

Closes #303.

(CRAFT-3675)

### CODE REVIEW

File: craft_cli/helptexts.py [hunk 1]
```diff
@@ -325,6 +326,9 @@ def _build_plain_command_help(
         )
 
         overview = textwrap.indent(command.overview, "    ")
+        # Remove reST-style double backticks
+        # Match _only_ double backticks, never triples
+        overview = re.sub(r"(?<!`)``(?!`)", "", overview)
```
Comment (lengau) [inline craft_cli/helptexts.py:331]: Normally I would suggest that we `re.compile` this into a global constant, but I think in this case this method should only ever run once per process invocation so it's probably not worth it?

**Review by lengau — APPROVED** [review]
Nice, thanks!

File: craft_cli/helptexts.py (hunk 1 above)
Comment (bepri) [reply craft_cli/helptexts.py:331]: I think it actually might make a separate call to this function for each field it's validating, so I'll move it to a constant. Good catch!

**Review by tigarmo — APPROVED** [review]
thanks!

**Pull request merged**